### PR TITLE
Try pcov for code coverage

### PR DESCRIPTION
**Pull request by staabm**
Similar to https://github.com/rectorphp/rector/pull/2576

Related article from the same source.. lets try

https://kizu514.com/blog/pcov-is-better-than-phpdbg-and-xdebug-for-code-coverage/
https://github.com/krakjoe/pcov

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -136,7 +136,7 @@ jobs:
             php: 7.3
             name: Test Coverage
             script:
-                - phpdbg -qrr -d memory_limit=-1 vendor/bin/phpunit --coverage-clover coverage.xml
+                - pecl install pcov; composer require pcov/clobber; vendor/bin/pcov clobber; vendor/bin/phpunit --configuration phpunit.xml --coverage-clover coverage.xml
```
Comment (TomasVotruba) [inline .travis.yml]: should be in standalone lines instead of `x; y;`

Comment (TomasVotruba) [reply .travis.yml]: Also reference to the post would be great, as I've never heard of it and it might be difficult to debug it in a year

Comment (staabm) [reply .travis.yml]: Will wait for the actual perf number and adjust in case its faster

**Comment by staabm** [comment]
What do you think about moving the simple checks to GithubAction so overall build-time can be reduced?

File: .travis.yml (hunk 1 above)
Comment (staabm) [reply .travis.yml]: Coverage is down to ~1min30

Comment (TomasVotruba) [reply .travis.yml]: Nice, go for it then :+1:

Comment (TomasVotruba) [reply .travis.yml]: Better remove fatal errors though: https://travis-ci.org/rectorphp/rector/jobs/633259038#L547

**Comment by TomasVotruba** [comment]
> What do you think about moving the simple checks to GithubAction so overall build-time can be reduced? (Less sequential tasks)

PR prototype would help

File: .travis.yml [hunk 2]
```diff
@@ -136,7 +136,9 @@ jobs:
             php: 7.3
             name: Test Coverage
             script:
-                - phpdbg -qrr -d memory_limit=-1 vendor/bin/phpunit --coverage-clover coverage.xml
+                # https://kizu514.com/blog/pcov-is-better-than-phpdbg-and-xdebug-for-code-coverage/
+                - pecl install pcov
+                - vendor/bin/phpunit --configuration phpunit.xml --coverage-clover coverage.xml
```
Comment (TomasVotruba) [inline .travis.yml]: Btw, why `--configuration phpunit.xml`? It's default value

Comment (staabm) [reply .travis.yml]: Copied from the article :)

**Comment by staabm** [comment]
Remaining failures seem unrelated

**Comment by TomasVotruba** [comment]
Indeed. Thanks :+1:

**Comment by TomasVotruba** [comment]
The coverage just dropped by 4 %

https://coveralls.io/github/rectorphp/rector?branch=master

image

That's way too much for 1 min improvements on parallel Travis run.

We'll need to revert

**Comment by staabm** [comment]
Imo the coverage did not actually change but pcov treats some details differenz and therefore leads to a different number

**Comment by TomasVotruba** [comment]
The number changed for much worse. If that would be ~0,5 % than it's ok.
But this might be seriously missleading. I'd rather stay closer to original xdebug coverage, as we'll probably get back to it in 2.9

Thanks for you work though. I wish I noticed the number before

**Pull request merged**